### PR TITLE
Swapd: Abort all tasks on FinishB

### PR DESCRIPTION
**Pull request by TheCharlatan**

### CODE REVIEW

File: src/swapd/runtime.rs [hunk 1]
```diff
@@ -1480,6 +1480,15 @@ impl Runtime {
...
                         self.state_update(senders, State::Bob(BobState::FinishB))?;
+                        let abort_all = Task::Abort(Abort {
+                            task_target: TaskTarget::AllTasks,
+                        });
+                        senders.send_to(
+                            ServiceBus::Ctl,
+                            self.identity(),
+                            self.syncer_state.monero_syncer(),
+                            Request::SyncerTask(abort_all),
+                        )?;
```
Comment (zkao) [inline src/swapd/runtime.rs:1491]: This would abort all task, if SweepSuccess were emitted, but it does not seem to be the case

**Review by zkao — APPROVED** [review]

File: src/swapd/runtime.rs (hunk 1 above)
Comment (TheCharlatan) [reply src/swapd/runtime.rs:1491]: Mmh, it worked fine for me. I will test it again.

Comment (TheCharlatan) [reply src/swapd/runtime.rs:1491]: Yeah, seems to be working. Can you test it again? If the problem persists, can you send me logs with the daemons running on debug?

Comment (zkao) [reply src/swapd/runtime.rs:1491]:
```
[2021-11-20T17:13:29Z INFO  farcaster_node::swapd::runtime] tx AccLock mined with 10 confirmations
[2021-11-20T17:14:42Z INFO  farcaster_node::swapd::runtime] Monero new height 969192
[2021-11-20T17:14:43Z INFO  farcaster_node::swapd::runtime] sweeping monero
[2021-11-20T17:14:44Z INFO  farcaster_node::syncerd::monero_syncer] sweeping to address: SweepAddress { id: 12, lifetime: 969681, addendum: Monero(SweepXmrAddress { spend_key: PrivateKey { scalar: Scalar{
    	bytes: [17, 232, 94, 197, 216, 203, 222, 74, 111, 41, 48, 146, 152, 48, 165, 51, 253, 108, 71, 178, 28, 52, 52, 234, 223, 123, 117, 244, 18, 17, 48, 11],
    } }, view_key: PrivateKey { scalar: Scalar{
    	bytes: [151, 174, 39, 225, 97, 47, 7, 45, 253, 143, 22, 175, 213, 190, 39, 18, 35, 14, 98, 78, 19, 228, 29, 44, 9, 155, 111, 168, 195, 146, 21, 3],
    } }, address: Address { network: Stagenet, addr_type: SubAddress, public_spend: 7f03695ca4261f47e18a697247a6de4df67210f48e96308e32673867e71122be, public_view: 6ea6307926329a27439b2df081bf919efbd1204dfd6427f9fef29db4a896a5f5 } }) }
[2021-11-20T17:14:50Z WARN  farcaster_node::syncerd::monero_syncer] error opening to be sweeped wallet: Server error: Failed to open wallet, falling back to generating a new wallet
[2021-11-20T17:14:55Z INFO  farcaster_node::syncerd::monero_syncer] sweeping address with balance: BalanceData { balance: 150000000, multisig_import_needed: false, per_subaddress: [SubaddressBalanceData { address: Address { network: Stagenet, addr_type: Standard, public_spend: baf4636d821d6e78976ed4f3b7399687c5b3cb812a666a65a9cd66848291987f, public_view: 46a397f42ca2d23c56e3fe671e33e9729a4a12a6147252438c246c1955517fa3 }, address_index: 0, balance: 150000000, label: "Primary account", num_unspent_outputs: 1, unlocked_balance: 150000000 }], unlocked_balance: 150000000 }
[2021-11-20T17:14:55Z INFO  farcaster_node::syncerd::monero_syncer] sweep transaction hash 818cfc67b3bc4ca6aeef1dd1256e120fa92564b4b908ed1788d1886d42dd9d3b
```

Comment (TheCharlatan) [reply src/swapd/runtime.rs:1491]: That just shows me it's working as intended up until that point. I need them on DEBUG and with the content after the line you are posting here.

Comment (zkao) [reply src/swapd/runtime.rs:1491]: [alice_maker.log](https://github.com/farcaster-project/farcaster-node/files/7575005/alice_maker.log)
[bob_taker.log](https://github.com/farcaster-project/farcaster-node/files/7575006/bob_taker.log)

Comment (zkao) [reply src/swapd/runtime.rs:1491]: and i confirm sweep was successful and monero appeared on my external wallet

Comment (TheCharlatan) [reply src/swapd/runtime.rs:1497]: Looks like the event is arriving:

```
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request syncer_bridge_ev(swap<0x9cabe315777b62fedcbc8b68747ddd431b0d981eb28ccc7b6ecbc4e10e108eb0>, SweepSuccess(SweepSuccess { id: 12, txids: [[106, 239, 159, 247, 170, 212, 166, 42, 56, 105, 19, 238, 68, 92, 202, 6, 167, 181, 11, 73, 24, 167, 233, 36, 225, 89, 25, 246, 116, 66, 229, 4]] })) from syncer<Monero> forwarded to processing
[2021-11-20T18:06:20Z DEBUG farcaster_node::syncerd::runtime] Syncerd BRIDGE RPC request: syncer_bridge_ev(swap<0x9cabe315777b62fedcbc8b68747ddd431b0d981eb28ccc7b6ecbc4e10e108eb0>, SweepSuccess(SweepSuccess { id: 12, txids: [[106, 239, 159, 247, 170, 212, 166, 42, 56, 105, 19, 238, 68, 92, 202, 6, 167, 181, 11, 73, 24, 167, 233, 36, 225, 89, 25, 246, 116, 66, 229, 4]] }))
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request processing complete
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request syncer_event(SweepSuccess(SweepSuccess { id: 12, txids: [[106, 239, 159, 247, 170, 212, 166, 42, 56, 105, 19, 238, 68, 92, 202, 6, 167, 181, 11, 73, 24, 167, 233, 36, 225, 89, 25, 246, 116, 66, 229, 4]] })) will be routed from syncer<Monero> to swap<0x9cabe315777b62fedcbc8b68747ddd431b0d981eb28ccc7b6ecbc4e10e108eb0>
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request processing complete
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request syncer_event(SweepSuccess(SweepSuccess { id: 12, txids: [[106, 239, 159, 247, 170, 212, 166, 42, 56, 105, 19, 238, 68, 92, 202, 6, 167, 181, 11, 73, 24, 167, 233, 36, 225, 89, 25, 246, 116, 66, 229, 4]] })) from syncer<Monero> forwarded to processing
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request processing complete
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request success(State transition: BobState(BuyProcSig) -> BobState(Finish)) from swap<0x9cabe315777b62fedcbc8b68747ddd431b0d981eb28ccc7b6ecbc4e10e108eb0> forwarded to processing
[2021-11-20T18:06:20Z DEBUG farcaster_node::farcasterd::runtime] processed all cli notifications
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request processing complete
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request syncer_task(Abort(Abort { task_target: AllTasks })) will be routed from swap<0x9cabe315777b62fedcbc8b68747ddd431b0d981eb28ccc7b6ecbc4e10e108eb0> to syncer<Monero>
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request processing complete
[2021-11-20T18:06:20Z DEBUG microservices::esb::controller] ESB request syncer_task(Abort(Abort { task_target: AllTasks })) from swap<0x9cabe315777b62fedcbc8b68747ddd431b0d981eb28ccc7b6ecbc4e10e108eb0> forwarded to processing
```

So all that is needed is targeting the bitcoin syncer as well, which I forgot. Done in c74abddba54c57c2e936d04503ed2d9e6ec5e7a1 .

**Review by zkao — APPROVED** [review]

**Pull request merged**